### PR TITLE
Limit the max parameter count in the state machine

### PR DESCRIPTION
**Pull request by skyline75489**
## Summary of the Pull Request

Currently there's no limit how many parameters that a control sequence can have. This PR fixes that.

 
## PR Checklist
* [ ] Closes #xxx
* [X] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### CODE REVIEW

**Comment by skyline75489** [comment]
I remember seeing this topic in one of @j4james 's comments but I can not find it..

File: src/terminal/parser/stateMachine.hpp [hunk 1]
```diff
@@ -27,6 +27,8 @@ namespace Microsoft::Console::VirtualTerminal
     // but for now 32767 is the safest limit for our existing code base.
     constexpr size_t MAX_PARAMETER_VALUE = 32767;
 
+    constexpr size_t MAX_PARAMETER_COUNT = 16384;
```
Comment (skyline75489) [inline src/terminal/parser/stateMachine.hpp:30]: The number is purely a random choice by myself, that I feel is somewhat reasonable.

**Comment by j4james** [comment]
I'm actually in the middle of refactoring the entire parameter handling system at the moment (and that includes a limit on the parameter count). If my proposed changes are accepted, then this PR is probably going to end up being a bit pointless.

I'm hoping to be finished soonish, but I don't want to make any promises, so you may still want to proceed with this in the meantime. I should mention, though, that I was going to set the limit to something more like 32. DEC recommends at least 16 (if I remember correctly), and I think most terminal emulators settled on double that value.

**Comment by skyline75489** [comment]
I'll close this for now in favor of what James will come up with.

**Pull request closed without merging**